project/CelebornBuild.scala

```diff
@@ -446,10 +446,12 @@ object Utils {
 }
 
 object CelebornSpi {
-  lazy val spi = Project("celeborn-spi", file("spi")).
-    settings(
+  lazy val spi = Project("celeborn-spi", file("spi"))
+    .settings(
       commonSettings,
-      releaseSettings
+      releaseSettings,
+      crossPaths := false,
+      Compile / doc / javacOptions := Seq("-encoding", UTF_8.name(), "-source", "1.8")
```

Comment:
Fix doc issue:
Before:
```
build/sbt "clean;celeborn-spi/doc"
Using /Users/fwang12/.sdkman/candidates/java/current as default JAVA_HOME.
Note, this will be overridden by -java-home if it is set.
[info] welcome to sbt 1.9.4 (Azul Systems, Inc. Java 1.8.0_392)
[info] loading settings for project celeborn-build from plugins.sbt ...
[info] loading project definition from /Users/fwang12/todo/celeborn/project
[info] compiling 1 Scala source to /Users/fwang12/todo/celeborn/project/target/scala-2.12/sbt-1.0/classes ...
[info] loading settings for project celeborn from version.sbt ...
[info] resolving key references (12502 settings) ...
[info] set current project to celeborn (in build file:/Users/fwang12/todo/celeborn/)
[success] Total time: 0 s, completed Jul 24, 2024 11:43:09 PM
[info] Main Java API documentation to /Users/fwang12/todo/celeborn/spi/target/api...
Provided by Standard doclet:
  -d <directory>                   Destination directory for output files
  -use                             Create class and package usage pages
  -version                         Include @version paragraphs
  -author                          Include @author paragraphs
  -docfilessubdirs                 Recursively copy doc-file subdirectories
  -splitindex                      Split index into one file per letter
  -windowtitle <text>              Browser window title for the documentation
  -doctitle <html-code>            Include title for the overview page
  -header <html-code>              Include header text for each page
  -footer <html-code>              Include footer text for each page
  -top    <html-code>              Include top text for each page
  -bottom <html-code>              Include bottom text for each page
  -link <url>                      Create links to javadoc output at <url>
  -linkoffline <url> <url2>        Link to docs at <url> using package list at <url2>
  -excludedocfilessubdir <name1>:.. Exclude any doc-files subdirectories with given name.
  -group <name> <p1>:<p2>..        Group specified packages together in overview page
  -nocomment                       Suppress description and tags, generate only declarations.
  -nodeprecated                    Do not include @deprecated information
  -noqualifier <name1>:<name2>:... Exclude the list of qualifiers from the output.
  -nosince                         Do not include @since information
  -notimestamp                     Do not include hidden time stamp
  -nodeprecatedlist                Do not generate deprecated list
  -notree                          Do not generate class hierarchy
  -noindex                         Do not generate index
  -nohelp                          Do not generate help link
  -nonavbar                        Do not generate navigation bar
  -serialwarn                      Generate warning about @serial tag
  -tag <name>:<locations>:<header> Specify single argument custom tags
  -taglet                          The fully qualified name of Taglet to register
  -tagletpath                      The path to Taglets
  -charset <charset>               Charset for cross-platform viewing of generated documentation.
  -helpfile <file>                 Include file that help link links to
  -linksource                      Generate source in HTML
  -sourcetab <tab length>          Specify the number of spaces each tab takes up in the source
  -keywords                        Include HTML meta tags with package, class and member info
  -stylesheetfile <path>           File to change style of the generated documentation
  -docencoding <name>              Specify the character encoding for the output
[warn] Unexpected javac output: javadoc: error - invalid flag: -g.
[info] Usage: javadoc [options] [packagenames] [sourcefiles] [@files]
[info]   -overview <file>                 Read overview documentation from HTML file
[info]   -public                          Show only public classes and members
[info]   -protected                       Show protected/public classes and members (default)
[info]   -package                         Show package/protected/public classes and members
[info]   -private                         Show all classes and members
[info]   -help                            Display command line options and exit
[info]   -doclet <class>                  Generate output via alternate doclet
[info]   -docletpath <path>               Specify where to find doclet class files
[info]   -sourcepath <pathlist>           Specify where to find source files
[info]   -classpath <pathlist>            Specify where to find user class files
[info]   -cp <pathlist>                   Specify where to find user class files
[info]   -exclude <pkglist>               Specify a list of packages to exclude
[info]   -subpackages <subpkglist>        Specify subpackages to recursively load
[info]   -breakiterator                   Compute first sentence with BreakIterator
[info]   -bootclasspath <pathlist>        Override location of class files loaded
[info]                                    by the bootstrap class loader
[info]   -source <release>                Provide source compatibility with specified release
[info]   -extdirs <dirlist>               Override location of installed extensions
[info]   -verbose                         Output messages about what Javadoc is doing
[info]   -locale <name>                   Locale to be used, e.g. en_US or en_US_WIN
[info]   -encoding <name>                 Source file encoding name
[info]   -quiet                           Do not display status messages
[info]   -J<flag>                         Pass <flag> directly to the runtime system
[info]   -X                               Print a synopsis of nonstandard options and exit
[info] 1 error
[warn] javadoc exited with exit code 1
[error] sbt.inc.Doc$JavadocGenerationFailed
[error]         at sbt.inc.Doc$.sbt$inc$Doc$$$anonfun$cachedJavadoc$1(Doc.scala:51)
[error]         at sbt.inc.Doc$$anonfun$cachedJavadoc$2.run(Doc.scala:41)
[error]         at sbt.inc.Doc$.sbt$inc$Doc$$$anonfun$prepare$1(Doc.scala:62)
[error]         at sbt.inc.Doc$$anonfun$prepare$5.run(Doc.scala:57)
[error]         at sbt.inc.Doc$.go$1(Doc.scala:73)
[error]         at sbt.inc.Doc$.$anonfun$cached$5(Doc.scala:82)
[error]         at sbt.inc.Doc$.$anonfun$cached$5$adapted(Doc.scala:81)
[error]         at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:220)
[error]         at sbt.inc.Doc$.sbt$inc$Doc$$$anonfun$cached$1(Doc.scala:85)
[error]         at sbt.inc.Doc$$anonfun$cached$7.run(Doc.scala:68)
[error]         at sbt.Defaults$.$anonfun$docTaskSettings$4(Defaults.scala:2178)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error]         at sbt.Execute.work(Execute.scala:292)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]         at java.lang.Thread.run(Thread.java:750)
[error] (celeborn-spi / Compile / doc) sbt.inc.Doc$JavadocGenerationFailed
```

After:
```
build/sbt "clean;celeborn-spi/doc"

Using /Users/fwang12/.sdkman/candidates/java/current as default JAVA_HOME.
Note, this will be overridden by -java-home if it is set.
[info] welcome to sbt 1.9.4 (Azul Systems, Inc. Java 1.8.0_392)
[info] loading settings for project celeborn-build from plugins.sbt ...
[info] loading project definition from /Users/fwang12/todo/celeborn/project
[info] loading settings for project celeborn from version.sbt ...
[info] resolving key references (12502 settings) ...
[info] set current project to celeborn (in build file:/Users/fwang12/todo/celeborn/)
[success] Total time: 0 s, completed Jul 24, 2024 11:53:43 PM
[info] Main Java API documentation to /Users/fwang12/todo/celeborn/spi/target/api...
[info] Loading source file /Users/fwang12/todo/celeborn/spi/src/main/java/org/apache/celeborn/spi/authentication/PasswordCredential.java...
[info] Loading source file /Users/fwang12/todo/celeborn/spi/src/main/java/org/apache/celeborn/spi/authentication/TokenCredential.java...
[info] Loading source file /Users/fwang12/todo/celeborn/spi/src/main/java/org/apache/celeborn/spi/authentication/Credential.java...
[info] Loading source file /Users/fwang12/todo/celeborn/spi/src/main/java/org/apache/celeborn/spi/authentication/PasswdAuthenticationProvider.java...
[info] Loading source file /Users/fwang12/todo/celeborn/spi/src/main/java/org/apache/celeborn/spi/authentication/TokenAuthenticationProvider.java...
[info] Constructing Javadoc information...
[info] Standard Doclet version 1.8.0_392
[info] Building tree for all the packages and classes...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/org/apache/celeborn/spi/authentication/Credential.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/org/apache/celeborn/spi/authentication/PasswdAuthenticationProvider.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/org/apache/celeborn/spi/authentication/PasswordCredential.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/org/apache/celeborn/spi/authentication/TokenAuthenticationProvider.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/org/apache/celeborn/spi/authentication/TokenCredential.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/org/apache/celeborn/spi/authentication/package-frame.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/org/apache/celeborn/spi/authentication/package-summary.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/org/apache/celeborn/spi/authentication/package-tree.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/constant-values.html...
[info] Building index for all the packages and classes...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/overview-tree.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/index-all.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/deprecated-list.html...
[info] Building index for all classes...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/allclasses-frame.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/allclasses-noframe.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/index.html...
[info] Generating /Users/fwang12/todo/celeborn/spi/target/api/help-doc.html...
[info] Main Java API documentation successful.
[success] Total time: 1 s, completed Jul 24, 2024 11:53:44 PM

```